### PR TITLE
Case recording and reading docs overhaul

### PR DESCRIPTION
**Pull request by ryanfarr01**
Additionally switched 'component_metadata' to 'component_options'

### CODE REVIEW

File: openmdao/docs/features/recording/saving_data.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+Instantiating a Recorder
+++++++++++++++++++++++++
+
+Instantiating a recorder is easy.  Simply give it a name, choose which type of recorder you want (currently only
+SqliteRecorder exists), and name the output file that you would like to write to.
+
+.. code-block:: console
+
+    self.my_recorder = SqliteRecorder("filename")
```
Comment (swryan) [inline openmdao/docs/features/recording/saving_data.rst]: probably didn't want `self` here

File: openmdao/recorders/tests/test_sqlite_reader.py [hunk 2]
```diff
@@ -1162,6 +1162,332 @@ def test_feature_load_system_case_for_restart(self):
...
+        first_driver_case = cr.driver_cases.get_case(0)
+        recorded_objectives = first_driver_case.get_objectives().keys
+        recorded_constraints = first_driver_case.get_constraints().keys
+        recorded_desvars = first_driver_case.get_desvars().keys
+
+        self.assertEqual(set(recorded_objectives), {'obj'})
+        self.assertEqual(set(recorded_constraints), {'con1', 'con2'})
+        self.assertEqual(set(recorded_desvars), {'x', 'z'})
+
+    def test_system_options(self):
```
Comment (swryan) [inline openmdao/recorders/tests/test_sqlite_reader.py]: looks like a duplicate of same `test_system_options` above

File: openmdao/docs/basic_guide/basic_recording.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+To record, all you have to do is create a case recorder (currently OpenMDAO only supports a SQLite case recorder),
+then attach it to any drivers, systems, or solvers whose data you want to store. This
+recorder creates a local SQLite database file when the optimization is run.
+
+.. code::
+
+    recorder = SqliteRecorder(case_filename)
+    p.driver.add_recorder(recorder)
+
+Create a Case Reader and Reading Data
```
Comment (kmarsteller) [inline openmdao/docs/basic_guide/basic_recording.rst]: "Creating" to match "Reading" and to match "Creating" above

File: openmdao/docs/features/recording/getting_metadata.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    :layout: interleave
+
+.. note::
+    Each system object must have a recorder explicitly attached in order for its metadata and options to be recorded.
+
+Variable Metadata
+-----------------
+
+Variable metadata is also made available through the CaseReader in :code:`output2meta` and :code:`input2meta`.
+For each variable the 'units', 'type', 'explicit', 'lower', and 'upper' are stored. note that this is recorded
```
Comment (kmarsteller) [inline openmdao/docs/features/recording/getting_metadata.rst]: "note" starts a new sentence, should be capitalized to "Note"

File: openmdao/docs/features/recording/iterating_hierarchically.rst [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+.. code-block:: console
+
+    case_keys = cr.driver_cases.list_cases()
+    for case_key in case_keys:
+        case = cr.driver_cases.get_case(case_key)
+        ...
+
+The Case Reader also has a :code:`get_cases` method, which provides a way to iterate
+over Driver and Solver cases in order with a choice of iterating hierarchically or flat
```
Comment (kmarsteller) [inline openmdao/docs/features/recording/iterating_hierarchically.rst]: "a choice of iterating either hierarchically or flaty" ?  What do you think? "Iterating in either a hierarchical or a flat manner" ?

**Pull request merged**